common/src/main/java/io/druid/common/config/Log4jShutdown.java

```diff
@@ -95,21 +96,22 @@ public void start()
   @Override
   public void stop()
   {
-    if (callbacksRun.get()) {
+    if (!state.compareAndSet(State.STARTED, State.STOPPING)) {
+      if (!state.waitTransition(State.STOPPING, State.STOPPED, SHUTDOWN_WAIT_TIMEOUT)) {
```

Comment:
is there a race here ? by the time waitTransition is called the state can be already stopped ?